### PR TITLE
Feature/wire up sso helper

### PR DESCRIPTION
**Pull request by jimmypuckett**

### CODE REVIEW

**Comment by jimmypuckett** [comment]
@ssfinney This is ready to merge so that we can tag version 1.0.0.  Can you look it over with me after Srum this morning?

File: readme.md [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 ## Prerequisite
 
-As side from Laravel >= 5.1, there is 1 package that is required.
+As side from Laravel >= 5.2, there is 1 package that is required.  NOTE: There are limitations in the way that 5.1 deals with collections, so I am taking the easy way & just not supporting it.  If you need this in 5.1, then please make a merge request on the `SsoController`.
```
Comment (ssfinney) [inline readme.md]: Aside

File: readme.md [hunk 2]
```diff
@@ -48,8 +48,86 @@ Add the Service Provider to `config/app.php`:
 
 All of the configuration values are stored in under a `discourse` key in `config/services.php`.  Here is the array to add...
 
-TODO: Put all of the config stuff here
+```php
+    'discourse' => [
+        // Entrypoint for Discourse to start the SSO process
```
Comment (ssfinney) [inline readme.md]:
```php
// The route's URI that acts as the entrypoint for Discourse to start the SSO process.
// Used by Discourse to route incoming logins.
```

File: src/Controllers/SsoController.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+use Illuminate\Contracts\Config\Repository as Config;
+use Illuminate\Http\Request;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Collection;
+
+/**
+ * Class SsoController
+ *
+ * Controller to process the Discourse SSO request.  There is a good bit of logic in here that almost feels like too
+ * much for a controller, but given that this is the only thing tht this controller is doing, I am not going to break
```
Comment (ssfinney) [inline src/Controllers/SsoController.php]: "thing that this"

File: readme.md [hunk 4]
```diff
@@ -17,7 +17,7 @@
 
 ## Prerequisite
 
-As side from Laravel >= 5.1, there is 1 package that is required.
+Aside from Laravel >= 5.2, there is 1 package that is required.  NOTE: There are limitations in the way that 5.1 deals with collections, so I am taking the easy way & just not supporting it.  If you need this in 5.1, then please make a merge request on the `SsoController`.
 
 * ["cviebrock/discourse-php": "^0.9.3",](https://github.com/briannesbitt/Carbon)
```
Comment (ssfinney) [inline readme.md]: Wrong package URL

**Pull request merged**